### DISCUSSION TITLE
how to set the initial trajectory seed for TrajOpt?

### DISCUSSION THREAD

**Discussion by xibeisiber** [tesseract-robotics/tesseract_ros#147] [General]
Hi all,
how can I set the initial trajectory seed for TrajOpt in its iteration process?
Is the default initial trajectory linear with possible collision?
I want to use ompl to generate a collision-free path first, and then feed it into the TrajOpt algorithm for optimization.

Thanks for any help!

**Comment by Levi-Armstrong** [comment]
I recommend taking a look at this [example](https://github.com/ros-industrial-consortium/tesseract_ros/blob/master/tesseract_ros_examples/src/freespace_ompl_example.cpp). It uses the FREESPACE_PLANNER process planner which does what you are requesting by first using ompl to generate an initial trajectory followed by running it through trajopt and time parameterization. The example uses the libraries directly but you can then take a look at the [tesseract_ros_workcell](https://github.com/Levi-Armstrong/tesseract_ros_workcell) to see have to leverage this in distributed fashion.
